### PR TITLE
Timestamp plugin

### PR DESCRIPTION
**Pull request by cflipse**
Adding a plugin for configuring commands to auto-timestamp.

This could _probably_ go into ROM core, as there's nothing database specific about it, but I'd like to iron out any kinks in a place where it's ... nominally expected.

I'm writing out the time explicitly in UTC, but I'm not great with timezone issues, so if anyone wants to point out the spots where I'm being blind here, that'd be great.

### CODE REVIEW

**Comment by cflipse** [comment]
So, it appears that this is failing CI on JRuby when run against SQLite, becase SQLite does not have a time datatype ... I presume that this _should_ be handled by a lower level adapter, and may be a bug in, say, sequel -- but I'm not sure it's really something that can be dealt with at that level?

I _could_ make the tests pass by just accounting for the case where the database fails to adapt the time _back_ into a Time object, but I feel like that might just be dodging a different problem...

**Comment by cflipse** [comment]
Also:  I'm considering also adding a `datestamp` function, which sets the specified attributes to the current date:  `datestamp :completed_on`  ... 

alternatively, could do the much less intuitive approach of not-handling the dates automatically:   `datestamp(:completed_on){ Date.today }`  ... which, however, ends up feeling much more like a "default attribute fill" plugin than a timestamp plugin (and could conceivably end up with not-matching times on multiple stamps ...)  ... think I've talked myself out of the block ;)

**Comment by solnic** [comment]
> So, it appears that this is failing CI on JRuby when run against SQLite, becase SQLite does not have a time datatype ... I presume that this should be handled by a lower level adapter, and may be a bug in, say, sequel -- but I'm not sure it's really something that can be dealt with at that level?

Sequel doesn't convert this for you because sqlite doesn't support time/date column types:

```
irb(main):010:0> db = Sequel.connect('jdbc:sqlite::memory')
=> #<Sequel::JDBC::Database: "jdbc:sqlite::memory">
irb(main):011:0> db.create_table(:notes) { primary_key :id; column :created_at, DateTime }
=> nil
irb(main):012:0> db[:notes].insert(created_at: DateTime.now)
=> 1
irb(main):013:0> db[:notes].first
=> {:id=>1, :created_at=>"2016-12-06 15:02:47.056000"}
irb(main):014:0> Note = Class.new(Sequel::Model(:notes))
=> Note
irb(main):015:0> Note.first
=> #<Note @values={:id=>1, :created_at=>"2016-12-06 15:02:47.056000"}>
```

**Comment by cflipse** [comment]
So, why does it pass on rubies not named `jquery` ?

**Comment by solnic** [comment]
They use a different driver (jruby uses jdbc-sqlite3)

**Comment by cflipse** [comment]
hah!  So it _is_ a driver thing, just much lower level. ;)

I'll tweak the spec to check against a string version.

**Comment by solnic** [comment]
Yeah it is, though I gotta say this discrepancy in behavior is pretty bad, but I'm not going to worry about it (at least not now) :)

**Comment by cflipse** [comment]
Fixed spec, added a `datestamp`

File: spec/integration/plugins/timestamp_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  include_context 'relations'
+
+  with_adapters do
+    context "with Create command" do
+
+    end
+
+  end
+
+end
```
Comment (solnic) [inline spec/integration/plugins/timestamp_spec.rb]: I suppose this can be deleted?

File: spec/unit/plugin/timestamp_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      expect(result).to include(:created_at).and(include(:updated_at)).and(include(:written))
+    end
+
+    it "sets timestamps on multi-tuple inputs" do
+      input = [{text: "note one"}, {text: "note two"}]
+
+      results = container.command(:notes).create_many.call(input)
+
+      results.each do |result|
+        expect(result).to include(:created_at).and(include(:updated_at))
```
Comment (solnic) [inline spec/unit/plugin/timestamp_spec.rb]: Does this check if these values are filled?

Comment (cflipse) [inline spec/unit/plugin/timestamp_spec.rb]: Mostly checks that the keys are present, which implies that ... _something_ has been done.  But, no, doesn't verify the times are filled.  dodging time comparisons mostly.

Will update.

Comment (solnic) [inline spec/unit/plugin/timestamp_spec.rb]: Yeah these keys will always be there, so it's important to check if we get back date time instances :)

File: spec/integration/plugins/timestamp_spec.rb (hunk 1 above)
Comment (cflipse) [inline spec/integration/plugins/timestamp_spec.rb]: Er, yep.

I'm not actually sure about the split between Unit an Integration in these cases; I ended up following one of the _other_ plugin specs and building into unit, but ... I tend to feel like things that actually hit the database a Really Integration Tests.  

:shrug:

File: spec/unit/plugin/timestamp_spec.rb (hunk 2 above)
Comment (cflipse) [inline spec/unit/plugin/timestamp_spec.rb]: Did the parsing song & dance, and checked that the time is correct within a second.  that _should_ be good ... I'm hesitant to want to drag in something like `timecop` _just_ so one can do time comparisons.  :roll_eyes:

**Comment by cflipse** [comment]
will rework this to use the `before` methods

**Comment by solnic** [comment]
Closing in favor of #135

**Pull request closed without merging**